packages/react-dom/src/__tests__/ReactDOMServerIntegrationLegacyContextDisabled-test.internal.js

```diff
@@ -105,7 +104,7 @@ describe('ReactDOMServerIntegrationLegacyContextDisabled', () => {
           <RegularFn />
         </span>
       </LegacyProvider>,
-      render === clientRenderOnBadMarkup ? 4 : 3,
+      3,
```

Comment:
On hydration mismatch we retry with client rendering in the new root so you got the warning again since it wasn't deduped. 

That we now have the same count with good and bad markup shows that it's properly deduped 👍🏻 

I'll add a comment to the original PR. There were a bunch of PR with the same pattern back then so I just forgot to mention it in each one.